crates/biome_html_formatter/src/utils/children.rs

```diff
@@ -236,9 +238,43 @@ where
                         }
                     }
                 }
+                prev_was_content = true;
             }
             child => {
+                let text = child.to_string();
+                let mut chunks = HtmlSplitChunksIterator::new(&text).peekable();
+
+                // Text starting with a whitespace
+                if let Some((_, HtmlTextChunk::Whitespace(_whitespace))) = chunks.peek() {
+                    match chunks.next() {
```

Comment:
Nit: you could `unwrap()` because it's a safe operation since the peek is already `Some`, and add a `// SAFETY: ` comment.

With this, you have less branching, and we do this in other areas of the code